src/main/java/com/teragrep/lsh_01/pool/RelpConnectionPool.java

```diff
@@ -35,17 +36,31 @@ public class RelpConnectionPool implements AutoCloseable {
 
     private final Supplier<IRelpConnection> relpConnectionWrapSupplier;
 
+    private final int rebindRequestAmount;
+
+    private final boolean rebindEnabled;
+
     private final ConcurrentLinkedQueue<IRelpConnection> queue;
 
+    // connection, how many times it has been used
+    private final HashMap<IRelpConnection, Integer> connectionsTaken;
```

Comment:
wouldn't this rather be on the responsibility of the ManagedConnection? It should be named as recordsSent?